### ISSUE TITLE
first entry and closing bracket in nested objects get flagged with comma first

### ISSUE THREAD

**Issue by ThaJay** [eslint/eslint#8676]
* **ESLint Version:** v3.19.0
* **Node Version:** v6.10.3
* **npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**What did you do? Please include the actual source code causing the issue.**
```js
const obj = {
  a     : 'foo'
, b     : 'bar'
, nested: {
    an: 'foon' // Expected indentation of 2 spaces but found 4. (indent)
  , bn: 'barn'
  } // Expected indentation of 0 spaces but found 2. (indent)
}
```
**relevant rules**
```js
"comma-style": ["warn", "first"],
"key-spacing": [
  "warn",
  {
    "singleLine": {
      "beforeColon": false,
      "afterColon": false
    },
    "align": {
      "beforeColon": false,
      "afterColon": true,
      "on": "colon"
    }
  }
]
```

**What did you expect to happen?**
The first entry in a nested object should not be interpreted as incorrect when using comma first notation. It works for the first entry of an object (no nesting) but not any level deeper.

**Please show your full configuration:**

```js
// .eslintrc.js
const error = "error"
const never = "never"
const always = "always"
const off = "off"
const warn = "warn"

module.exports = {
  "parser": "babel-eslint",
  "extends": ["standard", "standard-react"],
  "plugins": [
    "react",
    "jsx-a11y",
    "import"
  ],
  "globals": {
    "React":true,
    "__DEV__":true,
    "WebSocket":true,
    "XMLHttpRequest":true,
    "jest":true,
    "describe":true,
    "beforeAll":true,
    "afterAll":true,
    "beforeEach":true,
    "test":true,
    "expect":true
  },
  "rules": {
    "semi": [error, never],
    "space-before-function-paren": [error, always],
    "comma-style": [warn, "first"],
    "object-curly-spacing": [error, never],
    "curly": [error, "multi"],
    "react/prop-types": off,
    "import/extensions":off,
    "no-multi-spaces": off,
    "no-multiple-empty-lines": off,
    "padded-blocks":off,
    "key-spacing": [
      warn,
      {
        "singleLine": {
          "beforeColon": false,
          "afterColon": false
        },
        "align": {
          "beforeColon": false,
          "afterColon": true,
          "on": "colon"
        }
      }
    ]
  }
}
```

**Comment by kaicataldo** [comment]
The issue here is actually with the `indent` rule. With the following configuration:
```json
{
  "rules": {
    "comma-style": ["error", "first"],
    "indent": ["error", 2]
  }
}
```

The current way to satisfy both rules is to do the following:
```js
var obj = {
  a: 'foo' // Note that in ESLint v4.0, the new indent rule enforces this as well
  , b: 'bar'
  , nested: {
    an: 'foon'
    , bn: 'barn'
  }
}
```

I've never worked in a codebase that uses this style and don't have a lot of familiarity with it - the resource I see come up when I search for it is [this gist](https://gist.github.com/isaacs/357981). It unfortunately doesn't have any examples for nested object (or objects with a linebreak between the opening bracket and the first key.

If this style were to be supported, we would have to change the indent rule to account for commas.

**Comment by ThaJay** [comment]
This does not work either:
```
const obj =
{ a     : 'foo' // Expected indentation of 2 spaces but found 0. (indent), There should be no space after '{'. (object-curly-spacing)
, b     : 'bar'
, nested:
  { an: 'foon'
  , bn: 'barn'
  }
}
```

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
